### PR TITLE
Pagination integration tests

### PR DESCRIPTION
**Pull request by mikkoh**
Added some integration tests for pagination. Noticed there was an issue where the same client would be used for two tests which was causing issues where the wrong default client was being used.

### CODE REVIEW

File: scripts/generator/graphql_generator/csharp/SDK/DefaultQueries.cs.erb [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace Shopify.Unity.SDK {
     /// </summary>
     public class DefaultQueries {
         public static readonly int MaxPageSize = 250;
-        public static readonly int MaxProductPageSize = 20;
+        public static readonly int MaxProductPageSize = 10;
```
Comment (sleroux) [inline scripts/generator/graphql_generator/csharp/SDK/DefaultQueries.cs.erb]: Is this change intended? Just making sure since it wasn't mentioned in the PR description.

Comment (mikkoh) [reply scripts/generator/graphql_generator/csharp/SDK/DefaultQueries.cs.erb]: Yes it's intended. I think 10 gives us room to grow while still not being a major hit for users. These integration tests really informed those numbers.

File: Assets/IntegrationTests/TestShopifyCollections.cs [hunk 2]
```diff
@@ -36,5 +37,41 @@ public class TestShopifyCollections : MonoBehaviour {
...
             Assert.IsTrue (waiter.IsStopped, Utils.MaxQueryMessage);
         }
+
+        [UnityTest]
+        public IEnumerator LoadAllCollections() {
+            float maxDuration = 6f;
+            int maxCollections = 13;
+            StoppableWaitForTime waiter = new StoppableWaitForTime (maxDuration);
+            ShopifyError errorsFromQueries = null;
+            List<Collection> AllCollections = new List<Collection>();
```
Comment (JakeCataford) [inline Assets/IntegrationTests/TestShopifyCollections.cs]: We should be using type inference here where possible

File: Assets/IntegrationTests/TestShopifyCollections.cs [hunk 3]
```diff
@@ -36,5 +37,41 @@ public class TestShopifyCollections : MonoBehaviour {
...
+
+            Action<string> loadPage = null;
+
+            loadPage = (string after) => {
```
Comment (JakeCataford) [inline Assets/IntegrationTests/TestShopifyCollections.cs]: This can be condensed to `var loadPage = (...) => ...`

File: Assets/IntegrationTests/TestShopifyCollections.cs [hunk 4]
```diff
@@ -36,5 +37,41 @@ public class TestShopifyCollections : MonoBehaviour {
...
+                    if (errors != null) {
+                        errorsFromQueries = errors;
+                        waiter.Stop();
+                    } else if (afterCollectionsLoaded == null || AllCollections.Count >= maxCollections) {
+                        waiter.Stop();
+                    } else {
+                        loadPage(afterCollectionsLoaded);
+                    }
+                }, after: after);
+            };
```
Comment (JakeCataford) [inline Assets/IntegrationTests/TestShopifyCollections.cs]: We could build this into the sdk at some point, the recursive paginator

File: Assets/IntegrationTests/TestShopifyProducts.cs [hunk 5]
```diff
@@ -46,17 +47,16 @@ public class TestShopifyProducts : MonoBehaviour {
 
             yield return waiter;
 
-            Assert.IsTrue (waiter.IsStopped, Utils.MaxQueryMessage);
+            Assert.IsTrue(waiter.IsStopped, Utils.MaxQueryMessage);
```
Comment (JakeCataford) [inline Assets/IntegrationTests/TestShopifyProducts.cs]: This is a comment on shipped code, but I noticed `MaxQueryMessage` is wrong. As per the docs this message is shown when the test fails:

> Verifies that the specified condition is true. The assertion fails if the condition is false. Displays a message if the assertion fails.

So that message should be something like "Query failed to run in x seconds" or whatever instead of "Query finished in 1 seconds"

File: Assets/IntegrationTests/TestShopifyProducts.cs [hunk 6]
```diff
@@ -84,9 +84,45 @@ public class TestShopifyProducts : MonoBehaviour {
                 "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUyODE0NzU=", "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUyODQyOTE="
             );
 
-            yield return Utils.GetWaitQuery ();
+            yield return waiter;
+
+            Assert.IsTrue(waiter.IsStopped, Utils.MaxQueryMessage);
```
Comment (JakeCataford) [inline Assets/IntegrationTests/TestShopifyProducts.cs]: I still think it should be the job of the waiter to fail the test if it times out, which would remove this boilerplate

File: Assets/IntegrationTests/TestShopifyCollections.cs (hunk 3 above)
Comment (mikkoh) [reply Assets/IntegrationTests/TestShopifyCollections.cs]: Unfortunately it can't because it's a recursive lambda if you don't drop it to two lines it means that `loadPage` doesn't exist at the time of declaration.

File: Assets/IntegrationTests/TestShopifyCollections.cs (hunk 2 above)
Comment (mikkoh) [reply Assets/IntegrationTests/TestShopifyCollections.cs]: Hmmm... Why's that?

File: Assets/IntegrationTests/TestShopifyCollections.cs (hunk 4 above)
Comment (mikkoh) [reply Assets/IntegrationTests/TestShopifyCollections.cs]: Yeah I was thinking about that to just return a lambda.

File: Assets/IntegrationTests/TestShopifyCollections.cs (hunk 2 above)
Comment (JakeCataford) [reply Assets/IntegrationTests/TestShopifyCollections.cs]: Type inference makes the code more concise, especially where it is clear from the declaration what type the resulting variable will be. I was looking at https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions in their implicit local var section

File: Assets/IntegrationTests/TestShopifyCollections.cs (hunk 3 above)
Comment (JakeCataford) [reply Assets/IntegrationTests/TestShopifyCollections.cs]: ahh, right

File: Assets/IntegrationTests/TestShopifyCollections.cs (hunk 2 above)
Comment (mikkoh) [reply Assets/IntegrationTests/TestShopifyCollections.cs]: Yeah. In this case it might be cool to do. I just hate using `var` on a function return because I think it makes the code less readable especially considering that this is a public repo.

File: Assets/IntegrationTests/TestShopifyProducts.cs (hunk 6 above)
Comment (mikkoh) [reply Assets/IntegrationTests/TestShopifyProducts.cs]: Yeah maybe. Let's leave for now and refactor later when there's time.
